pkg/channel/event_receiver_test.go

```diff
@@ -165,7 +215,7 @@ func TestEventReceiver_ServeHTTP(t *testing.T) {
 }
 
 func TestEventReceiver_ParseChannel(t *testing.T) {
-	url, _ := url.Parse("test-channel.test-namespace.svc.")
+	url, _ := url.Parse("http://test-channel.test-namespace.svc.")
```

Comment:
hrm...?

Reply:
The test itself uses url.Parse, which does not work properly for scheme-less URLs. That's what this whole CL is about!

Reply:
https://play.golang.org/p/4HGT_FQKyS3

Reply:
https://play.golang.org/p/1QzsUwJzJTj check this one out and look at the fields populated by Parse. Also golang documentation can help too! Given string "thisworks" you want that to be the host. Given "thisworks/abc", you want "thisworks" to be the host and "/abc" to be the path. The reason you definitely want this is that "//thisworks" will make "thisworks" the host. Currently the parsing checked in works for scheme-less URLs only and will break (because we use the url.Path field everywhere) if an actual URL with a scheme is provided. Does that clarify this a bit?

Reply:
aaah ok thanks for the other playground. that clears it up a lot!!